storage/innobase/btr/btr0pcur.cc

```diff
@@ -260,16 +257,19 @@ static bool btr_pcur_optimistic_latch_leaves(btr_pcur_t *pcur,
   }
 
   buf_block_t *prev;
+  const page_t *prev_page= nullptr;
   if (left_page_no != FIL_NULL)
   {
     prev= buf_page_get_gen(page_id_t(pcur->old_page_id.space(),
                                      left_page_no), block->zip_size(),
                            mode, nullptr, BUF_GET_POSSIBLY_FREED, mtr);
-    if (!prev ||
-        page_is_comp(prev->page.frame) != page_is_comp(block->page.frame) ||
-        memcmp_aligned<2>(block->page.frame, prev->page.frame, 2) ||
-        memcmp_aligned<2>(block->page.frame + PAGE_HEADER + PAGE_INDEX_ID,
-                          prev->page.frame + PAGE_HEADER + PAGE_INDEX_ID, 8))
+    if (!prev)
+      goto fail;
+    prev_page= prev->page.frame;
```

Comment:
If I change `btr_pcur_get_page()` and friends to refer directly to `block->page.frame`, then a non-inlined version of this function would be the same size. But, inlined as part of `btr_pcur_t::restore_position` the result would be 20 bytes larger again. Once I reverted the changes related to the variable `prev_page` the size would be the same again.

I must also remove any added use of `page_align()` and `page_offset()` elsewhere.

Reply:
It turns out that we need to add `page` parameters to many functions that currently only take `rec`. I am working on that. It will be a huge change, so I may end up splitting this fix into two parts.